### ISSUE TITLE
Nimbus 24.1.0 SIGSEGV on checkpoint sync

### ISSUE THREAD

**Issue by yorickdowne** [status-im/nim-beacon-chain#5716]
**Describe the bug**

Nimbus 24.1.0 fails with SIGSEGV on checkpoint sync. 23.11.0 succeeds. Logs:

```
eth-docker-consensus-1  | NTC 2024-01-06 22:19:16.984+00:00 Starting trusted node sync                 databaseDir=/var/lib/nimbus/db backfill=false reindex=false syncTarget=finalized restUrl=https://beaconstate.info
eth-docker-consensus-1  | NTC 2024-01-06 22:19:16.984+00:00 Downloading checkpoint state               syncTarget=finalized restUrl=https://beaconstate.info stateId=finalized
eth-docker-consensus-1  | NTC 2024-01-06 22:19:20.069+00:00 Database initialized from genesis          blockRoot=4d611d5b93fdab69013a7f0a2f961caca0c853f87cfe9595fe50038163079360 stateRoot=7e76880eb67bbdc86250aa578958e9d0675e64e714337855204fb5abaaf82c2b stateSlot=0
eth-docker-consensus-1  | NTC 2024-01-06 22:19:34.825+00:00 Checkpoint written to database             blockRoot=4b617c07d2658fff4371050d65da946f103996c256547c4e4aee64020321700a stateRoot=62b17c9774a2df10923e677292a2d890481a9708fa20940042fcc258ad63b4b9 stateSlot=8146176
eth-docker-consensus-1  | INF 2024-01-06 22:19:35.313+00:00 Loading finalized blocks                   finHigh=none() finalizedHead=4b617c07:8146176
eth-docker-consensus-1  | INF 2024-01-06 22:19:35.523+00:00 Block DAG initialized                      head=4b617c07:8146176 finalizedHead=4b617c07:8146176 tail=4b617c07:8146176 backfill="(8146176, \"332dca98\")" loadDur=656us658ns summariesDur=365ms531us223ns finalizedDur=331ms544us503ns frontfillDur=296us719ns keysDur=26us741ns
eth-docker-consensus-1  | NTC 2024-01-06 22:19:35.523+00:00 Database initialized, historical blocks will be backfilled when starting the node missingSlots=1056768 backfill="(slot: 8146176, parent_root: 332dca98e31d13fe3f739fa5a998c8d91928c52be3baeadda741be9851142712)" horizon=7089408 syncTarget=finalized restUrl=https://beaconstate.info
eth-docker-consensus-1  | NTC 2024-01-06 22:19:35.523+00:00 Done, your beacon node is ready to serve you! Don't forget to check that you're on the canonical chain by comparing the checkpoint root with other online sources. See https://nimbus.guide/trusted-node-sync.html for more information. checkpoint=4b617c07:8146176 syncTarget=finalized restUrl=https://beaconstate.info
eth-docker-consensus-1  | INF 2024-01-06 22:19:35.811+00:00 Launching beacon node                      topics="beacnde" version=v24.1.0-00ba72-stateofus bls_backend=BLST cmdParams="@[\"--data-dir=/var/lib/nimbus\", \"--non-interactive\", \"--status-bar=false\", \"--tcp-port=9010\", \"--udp-port=9010\", \"--max-peers=160\", \"--el=http://execution:8551\", \"--jwt-secret=/var/lib/nimbus/ee-secret/jwtsecret\", \"--rest\", \"--rest-address=0.0.0.0\", \"--rest-port=5052\", \"--enr-auto-update=true\", \"--metrics\", \"--metrics-port=8008\", \"--metrics-address=0.0.0.0\", \"--suggested-fee-recipient=0xDccf8451070a86183eE70D330C4c43b686E9CF86\", \"--in-process-validators=false\", \"--network=mainnet\", \"--log-level=INFO\", \"--doppelganger-detection=false\", \"--history=prune\"]" config="(configFile: none(InputFile), logLevel: \"INFO\", logStdout: auto, logFile: none(OutFile), eth2Network: some(\"mainnet\"), dataDir: /var/lib/nimbus, validatorsDirFlag: none(InputDir), verifyingWeb3Signers: @[], provenBlockProperties: @[], web3Signers: @[], web3signerUpdateInterval: 3600, secretsDirFlag: none(InputDir), walletsDirFlag: none(InputDir), eraDirFlag: none(InputDir), web3ForcePolling: none(bool), web3Urls: @[], elUrls: @[(url: \"http://execution:8551\", jwtSecret: none(string), jwtSecretFile: none(InputFile), roles: none(EngineApiRoles))], noEl: false, optimistic: none(bool), requireEngineAPI: none(bool), nonInteractive: true, netKeyFile: \"random\", netKeyInsecurePassword: false, agentString: \"nimbus\", subscribeAllSubnets: false, slashingDbKind: v2, numThreads: 0, jwtSecret: some(/var/lib/nimbus/ee-secret/jwtsecret), cmd: noCommand, runAsServiceFlag: false, bootstrapNodes: @[], bootstrapNodesFile: , listenAddress: 0.0.0.0, tcpPort: 9010, udpPort: 9010, maxPeers: 160, hardMaxPeers: none(int), nat: (hasExtIp: false, nat: NatAny), enrAutoUpdate: true, enableYamux: false, weakSubjectivityCheckpoint: none(Checkpoint), externalBeaconApiUrl: none(string), syncLightClient: true, trustedBlockRoot: none(Hash256), trustedStateRoot: none(Hash256), finalizedCheckpointState: none(InputFile), genesisState: none(InputFile), genesisStateUrl: none(Uri), finalizedDepositTreeSnapshot: none(InputFile), finalizedCheckpointBlock: none(InputFile), nodeName: \"\", graffiti: none(GraffitiBytes), strictVerification: false, stopAtEpoch: 0, stopAtSyncedEpoch: 0, metricsEnabled: true, metricsAddress: 0.0.0.0, metricsPort: 8008, statusBarEnabled: false, statusBarContents: \"peers: $connected_peers;finalized: $finalized_root:$finalized_epoch;head: $head_root:$head_epoch:$head_epoch_slot;time: $epoch:$epoch_slot ($slot);sync: $sync_status|ETH: $attached_validators_balance\", rpcEnabled: none(bool), rpcPort: none(Port), rpcAddress: none(IpAddress), restEnabled: true, restPort: 5052, restAddress: 0.0.0.0, restAllowedOrigin: none(string), restCacheSize: 3, restCacheTtl: 60, restRequestTimeout: 0, restMaxRequestBodySize: 16384, restMaxRequestHeadersSize: 128, keymanagerEnabled: false, keymanagerPort: 5052, keymanagerAddress: 127.0.0.1, keymanagerAllowedOrigin: none(string), keymanagerTokenFile: none(InputFile), lightClientDataServe: true, lightClientDataImportMode: only-new, lightClientDataMaxPeriods: none(uint64), inProcessValidators: false, discv5Enabled: true, dumpEnabled: false, directPeers: @[], doppelgangerDetection: false, syncHorizon: 50, terminalTotalDifficultyOverride: none(string), validatorMonitorAuto: true, validatorMonitorPubkeys: @[], validatorMonitorDetails: false, validatorMonitorTotals: none(bool), safeSlotsToImportOptimistically: none(uint16), suggestedFeeRecipient: some(0xdccf8451070a86183ee70d330c4c43b686e9cf86), suggestedGasLimit: 30000000, payloadBuilderEnable: false, payloadBuilderUrl: \"\", localBlockValueBoost: 0, historyMode: prune, trustedSetupFile: none(string), bandwidthEstimate: none(Natural), forkChoiceVersion: none(ForkChoiceVersion))"
eth-docker-consensus-1  | NTC 2024-01-06 22:19:35.812+00:00 Starting metrics HTTP server               topics="beacnde" url=http://0.0.0.0:8008/metrics
eth-docker-consensus-1  | INF 2024-01-06 22:19:35.812+00:00 Threadpool started                         topics="beacnde" numThreads=12
eth-docker-consensus-1  | INF 2024-01-06 22:19:37.626+00:00 Loading block DAG from database            topics="beacnde" path=/var/lib/nimbus/db
eth-docker-consensus-1  | INF 2024-01-06 22:19:38.360+00:00 Block DAG initialized                      head=4b617c07:8146176 finalizedHead=4b617c07:8146176 tail=4b617c07:8146176 backfill="(8146176, \"332dca98\")" loadDur=2ms106us935ns summariesDur=375ms865us263ns finalizedDur=355ms775us65ns frontfillDur=371us803ns keysDur=38us781ns
eth-docker-consensus-1  | ERR 2024-01-06 22:19:38.360+00:00 Block failed to load unexpectedly          topics="chaindag_lc" bid=4b617c07:8146176 tail=8146176
eth-docker-consensus-1  | NTC 2024-01-06 22:19:38.360+00:00 Starting REST HTTP server                  topics="beacnde" url=http://0.0.0.0:5052
eth-docker-consensus-1  | INF 2024-01-06 22:19:38.361+00:00 Generating new networking key              topics="networking" network_public_key=08021221026cdb01c4f820289d356d2028153c6b8ec565de73f1378e73ec603551d055a744 network_peer_id=16Uiu2HAm2kbhgPs93usJB3mLHFiz5iudEke5XXDHPVVYAVvaoyVm
eth-docker-consensus-1  | WRN 2024-01-06 22:19:46.117+00:00 UPnP/NAT-PMP not available                 topics="eth net nat"
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.118+00:00 Discovery ENR initialized                  topics="eth p2p discv5" enrAutoUpdate=true seqNum=1 ip=none(IpAddress) tcpPort=some(9010) udpPort=some(9010) customEnrFields="@[eth2:0xBBA4DA9603000000FFFFFFFFFFFFFFFF, attnets:0x0000000000000000]" uri=enr:-Kq4QPIOibE261TFuI_FYDR5SCrLC_1LytwcwZKuW2SIdQCmdW5j09Um9hVeJIWC5umN92Pk4P2dz4lR9G2kCZzxah8Bh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC7pNqWAwAAAP__________gmlkgnY0iXNlY3AyNTZrMaECbNsBxPggKJ01bSAoFTxrjsVl3nPxN45z7GA1UdBVp0SDdGNwgiMyg3VkcIIjMg
eth-docker-consensus-1  | NTC 2024-01-06 22:19:46.118+00:00 No external IP provided for the ENR, this node will not be discoverable until the ENR is updated with the discovered external IP address topics="eth p2p discv5"
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.118+00:00 Loading slashing protection database (v2)  topics="beacnde" path=/var/lib/nimbus/validators
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.352+00:00 Initializing fork choice                   topics="beacnde" unfinalized_blocks=0
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.352+00:00 Fork choice initialized                    topics="beacnde" justified=254567:f7e9dbc9 finalized=254566:039015c8
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.353+00:00 Loading validators                         topics="beacval" validatorsDir=/var/lib/nimbus/validators keystore_cache_available=true
eth-docker-consensus-1  | NTC 2024-01-06 22:19:46.353+00:00 Starting beacon node                       topics="beacnde" version=v24.1.0-00ba72-stateofus nimVersion=1.6.16 enr=enr:-Kq4QPIOibE261TFuI_FYDR5SCrLC_1LytwcwZKuW2SIdQCmdW5j09Um9hVeJIWC5umN92Pk4P2dz4lR9G2kCZzxah8Bh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC7pNqWAwAAAP__________gmlkgnY0iXNlY3AyNTZrMaECbNsBxPggKJ01bSAoFTxrjsVl3nPxN45z7GA1UdBVp0SDdGNwgiMyg3VkcIIjMg peerId=16Uiu2HAm2kbhgPs93usJB3mLHFiz5iudEke5XXDHPVVYAVvaoyVm timeSinceFinalization=24m11s353ms158us628ns head=4b617c07:8146176 justified=254567:f7e9dbc9 finalized=254566:039015c8 finalizedHead=4b617c07:8146176 SLOTS_PER_EPOCH=32 SECONDS_PER_SLOT=12 SPEC_VERSION=1.4.0-beta.5 dataDir=/var/lib/nimbus validators=0
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.355+00:00 Listening to incoming network requests     topics="beacnde"
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.355+00:00 Starting discovery node                    topics="eth p2p discv5" node=72*a659ea:unaddressable bindAddress=0.0.0.0:9010
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.356+00:00 Starting execution layer deposit syncing   topics="elmon" contract=0x00000000219ab540356cbb839cbe05303d7705fa
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.356+00:00 Connection attempt started                 topics="elmon"
eth-docker-consensus-1  | NTC 2024-01-06 22:19:46.357+00:00 REST service started                       address=0.0.0.0:5052
eth-docker-consensus-1  | NTC 2024-01-06 22:19:46.357+00:00 Starting light client                      topics="lightcl" trusted_block_root=none(Hash256)
eth-docker-consensus-1  | INF 2024-01-06 22:19:46.357+00:00 Scheduling first slot action               topics="beacnde" startTime=161w4d10h19m23s357ms877us945ns nextSlot=8146297 timeToNextSlot=642ms122us55ns
eth-docker-consensus-1  | Traceback (most recent call last, using override)
eth-docker-consensus-1  | /home/user/nimbus-eth2/vendor/nim-libp2p/libp2p/protocols/rendezvous.nim(905) main
eth-docker-consensus-1  | /home/user/nimbus-eth2/vendor/nim-libp2p/libp2p/protocols/rendezvous.nim(898) NimMain
eth-docker-consensus-1  | /home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2340) main
eth-docker-consensus-1  | /home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2208) handleStartUpCmd
eth-docker-consensus-1  | /home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2109) doRunBeaconNode
eth-docker-consensus-1  | /home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1875) start
eth-docker-consensus-1  | /home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1822) run
eth-docker-consensus-1  | /home/user/nimbus-eth2/vendor/nim-chronos/chronos/internal/asyncengine.nim(146) poll
eth-docker-consensus-1  | /home/user/nimbus-eth2/vendor/nim-chronos/chronos/internal/asyncfutures.nim(359) futureContinue
eth-docker-consensus-1  | /home/user/nimbus-eth2/beacon_chain/el/el_manager.nim(2084) syncEth1Chain
eth-docker-consensus-1  | /home/user/nimbus-eth2/vendor/nimbus-build-system/vendor/Nim/lib/system/excpt.nim(631) signalHandler
eth-docker-consensus-1  | SIGSEGV: Illegal storage access. (Attempt to read from nil?)
eth-docker-consensus-1 exited with code 0
```

**Comment by nikicat** [comment]
The same for me on aarch64. 23.11.0 works well.

**Comment by tersec** [comment]
https://github.com/status-im/nimbus-eth2/pull/5695 and released in v24.1.1.
